fe/fe-core/src/main/java/com/starrocks/alter/AlterJobExecutor.java

```diff
@@ -617,8 +616,12 @@ public Void visitDropPartitionClause(DropPartitionClause clause, ConnectContext
             DynamicPartitionUtil.checkAlterAllowed((OlapTable) table);
         }
 
-        ErrorReport.wrapWithRuntimeException(() ->
-                GlobalStateMgr.getCurrentState().getLocalMetastore().dropPartition(db, table, clause));
+        try (AutoCloseableLock ignore =
+                    new AutoCloseableLock(new Locker(), db.getId(), Lists.newArrayList(table.getId()), LockType.WRITE)) {
+            ErrorReport.wrapWithRuntimeException(() ->
+                    GlobalStateMgr.getCurrentState().getLocalMetastore().dropPartition(db, table, clause));
```

Comment:
It's a little confusing that the `addPartitions()` automatically acquire the db/table inside the implementation, but not this `dropPartition` interface. This anti-symmetric implementation is so easy to lead incorrect usage.

Reply:
You are absolutely right. I think locking should be done at the execution layer, not the storage layer. Because this is a Lock, not a Latch, corresponding to a transaction. But unfortunately, we currently do not have a clear code logic layering.